handbook/repository/aws-secrets.md

```diff
@@ -0,0 +1,191 @@
+
+{% hint style="warning" %}
+The SSO login expires every 8 hours.
+{% endhint %}
+
+### Using AWS session
+
+This method is more manual where you will need to export environments or change file by yourself.
+
+You will need to go to your [AWS account](https://island-is.awsapps.com/start) and get the required credentials for the account you need.
+
+1. Option 1: Set environments variables
+
+You can first copy/paste theses environments variables on your terminal.
+
+```bash
+export AWS_ACCESS_KEY_ID=AWS_ACCESS_KEY_ID_EXAMPLE
+export AWS_SECRET_ACCESS_KEY=AWS_SECRET_ACCESS_KEY_EXAMPLE
+export AWS_SESSION_TOKEN=AWS_SESSION_TOKEN_EXAMPLE
+```
+
+2. Option 2: Edit `~/.aws/credentials`
+
+Or copy/paste the values in the `~/.aws/credentials` file.
+
+```bash
+[X_AWSPowerUserAccess]
+aws_access_key_id = <KEY_ID>
+aws_secret_access_key = <ACCESS_KEY>
+aws_session_token = <SESSION_TOKEN>
+```
+
+3. Refresh your profile
+
+You will need to log in to your AWS account and get the refreshed credentials, with one of the following method above.
+
+4. Ready to use
+
+In this case you won't need to pass a profile name as opposed to the SSO method.
+
+```bash
+yarn get-secrets <project-name>
+```
+
+{% hint style="warning" %}
+The session token expires every 1 hour.
+{% endhint %}
+
+## Usage to fetch secrets
+
+You should now be able to fetch secrets for the project you need.
+
+**With SSO**
+
+```bash
+AWS_PROFILE=<profile-name> yarn get-secrets <project> [options]
+```
+
+**Without SSO**
+
+```bash
+yarn get-secrets <project> [options]
+```
+
+**Example**:
+
+```bash
+yarn get-secrets api
+```
+
+You can verify it by opening the `.env.secret` file at the root, or inside your code using for example:
+
+```typescript
+const { MY_SECRET_KEY } = process.env
+```
+
+You can also add the `--reset` argument to the command, that will reset the .env.secret file.
+
+### Troubleshoot
+
+If you get the following error message, you will need to refresh your credentials as explained above.
+
+```bash
+An error occurred (ExpiredTokenException) when calling the GetParametersByPath operation: The security token included in the request is expired
+```
+
+## Usage to create secrets
+
+You can run the following command and will be prompted for input.
+
+```bash
+yarn create-secret
+```
+
+You will be asked for a _secret name_ that will be added to the `/k8s/` secrets namespace, a _secret value_ and the _secret type_ (`SecureString` or `String`).
+
+### Example
+
+```bash
+➜ yarn create-secret
+
+Secret name: /k8s/my-app/MY_APP_KEY
+# Name: Ok!
+# Length: Ok!
+
+Secret value: a-very-secure-secret
+# Length: Ok!
+
+SecureString [Y/n]? # [enter] for SecureString
+# SecureString selected
+
+Add tags? [y/N]? # [enter] to skip creating tags
+
+Example: Key=Foo,Value=Bar Key=Another,Value=Tag: # note: Key and Value are case sensitive! Create multiple tags by separating with whitespace.
```

Comment:
Would be nice if this script asked if you want to add the dev label (separate from tags) to allow devs to fetch the dev secret (should add this to the backlog).